python/cugraph/dask/link_analysis/pagerank.py

```diff
@@ -117,11 +133,14 @@ def pagerank(input_graph,
 
     client = default_client()
 
-    if(input_graph.local_data is not None and
-       input_graph.local_data['by'] == 'dst'):
-        data = input_graph.local_data['data']
-    else:
-        data = get_local_data(input_graph, by='dst', load_balance=load_balance)
+    input_graph.compute_renumber_edge_list(transposed=True)
+    (ddf,
+     num_verts,
+     partition_row_size,
+     partition_col_size,
+     vertex_partition_offsets) = shuffle(input_graph, transposed=True)
```

Comment:
Note: We can add prows and pcols optimization here i.e. shuffle(input_graph, prows=2, pcols=2, transposed=True)

Reply:
If we do, probably should expose it as optional parameters to pagerank as well, since the idea is to allow a sophisticated caller of pagerank to specify prows/pcols.

Reply:
Sophisticated users have options to re-initialize comms. Unless this turns out absolutely necessary, I am more leaning towards providing just one mechanism to control this to keep things simple.

Reply:
I'm not sure how one would take a configuration with 16 GPUs, given our current Python API and have it create a 2 x 8 grid of GPUs.  Unless I'm missing something in the current implementation.

The shuffle function computes prows and pcols if it not specified, and computes them to be as close to a square as they can be, given a collection of GPUs.  So with 16 GPUs, prows = 4 and pcols = 4 unless you specify otherwise in the call to shuffle.

Is the suggestion here that shuffle shouldn't do that but should instead use what the comms object gets initialized?  If so then we would need to update shuffle to eliminate these parameters and provide a useful example as to how to execute an MNMG algorithm with a custom GPU configuration.  Currently the only documented mechanism to control this is the parameters to shuffle.